src/main/java/io/dropwizard/revolver/provider/BlacklistProcessor.java

```diff
@@ -0,0 +1,88 @@
+package io.dropwizard.revolver.provider;
+
+import lombok.extern.slf4j.Slf4j;
+import org.glassfish.jersey.server.model.ModelProcessor;
+import org.glassfish.jersey.server.model.Resource;
+import org.glassfish.jersey.server.model.ResourceMethod;
+import org.glassfish.jersey.server.model.ResourceModel;
+
+import javax.ws.rs.core.Configuration;
+import javax.ws.rs.ext.Provider;
+import java.util.List;
+import java.util.Set;
+
+@Provider
+@Slf4j
+public class BlacklistProcessor implements ModelProcessor {
+
+    private final Set<BlacklistMethodData> blacklistMethods;
+
+    public BlacklistProcessor(final Set<BlacklistMethodData> blacklistMethods) {
+        this.blacklistMethods = blacklistMethods;
+    }
+
+    @Override
+    public ResourceModel processResourceModel(final ResourceModel resourceModel, final Configuration configuration) {
+        if (blacklistMethods != null && blacklistMethods.isEmpty()) {
+            log.info("No API end-point to blacklist");
+            return resourceModel;
+        }
+
+        ResourceModel.Builder newResourceModelBuilder = new ResourceModel.Builder(false);
+        for (final Resource resource : resourceModel.getResources()) {
```

Comment:
refactor to reduce cyclomatic complexity for the nested for and if branches. probably use java 8 streams and collectors with filters

Reply:
Streams would be difficult to use here.